src/Microsoft.AspNetCore.Sockets.Common/Internal/Formatters/ServerSentEventsMessageFormatter.cs

```diff
@@ -27,70 +30,53 @@ public static bool TryFormatMessage(Message message, Span<byte> buffer, out int
                 throw new InvalidOperationException("Cannot format message where endOfMessage is false using this format");
             }
 
-            // Need at least: Length of 'data: ', one character type, one \r\n, and the trailing \r\n
-            if (buffer.Length < DataPrefix.Length + 1 + Newline.Length + Newline.Length)
+            if (!TryGetTypeIndicator(message.Type, out var typeIndicator))
             {
-                bytesWritten = 0;
                 return false;
             }
-            DataPrefix.CopyTo(buffer);
-            buffer = buffer.Slice(DataPrefix.Length);
-            if (!TryFormatType(buffer, message.Type))
+
+            // Write the Data Prefix
+            if (!output.TryWrite(DataPrefix))
             {
-                bytesWritten = 0;
                 return false;
             }
-            buffer = buffer.Slice(1);
 
-            Newline.CopyTo(buffer);
-            buffer = buffer.Slice(Newline.Length);
+            // Write the type indicator
+            output.Append(typeIndicator, TextEncoder.Utf8);
```

Comment:
That's the thing I need to clarify about the `IOutput` API. It's guaranteed to continue trying to append and enlarging the buffer until it succeeds. But if the implementer of `IOutput` can't expand the buffer anymore, I don't know what the expected behavior is here yet.